### DISCUSSION TITLE
Should we remove circular dependency checking?

### DISCUSSION THREAD

**Discussion by msujew** [eclipse-theia/theia#11863] [Improvements] [answered]
I've been working on improving the dev-experience for building Theia recently (see https://github.com/eclipse-theia/theia/pull/11830 and https://github.com/eclipse-theia/theia/pull/11835). The last warnings that appear are from the circular dependency checking plugin for webpack. These generally indicate some level of code smell, though I would argue that it's no issue in our project, especially since webpack handles these issues gracefully.

We only use the circular dependency check plugin to provide us with warnings, which we generally tend to ignore. I've been wondering why we use the plugin at all, and only found commit 171be73, which was pretty early on in the project. When migrating to newer versions/workflows of webpack, we always seem to have brought it along, but it was never explicitly mentioned why it's there in the first place.

As it serves no real value for the project internally, I would propose to remove it from the generated webpack build. Downstream users can always choose to add the plugin to their own builds, but we shouldn't enforce it on them. First-time users of Theia often appear confused due to the errors/warnings our generated webpack build prints, so I would try to minimize them. I would like to hear the opinion of Theia consumers and other contributors on this topic.

**Comment by JonasHelming** [comment, marked as answer]
As discussed in the weekly dev call: Yes, let's remove this check!
